File: mobile/types/feedback.ts
export interface Feedback {
  id: string;
  meeting_id: string;
  speaker_id: string;
  evaluator_id: string;
  comment: string;
  created_at: string;
  updated_at: string;
}

export type EmailStatus = 'pending' | 'sent' | 'failed';

export interface EmailLog {
  id: string;
  meeting_id: string;
  speaker_id: string;
  sent_at: string | null;
  status: EmailStatus;
  error_message: string | null;
}

export interface EmailDispatchResult {
  sent: number;
  failed: number;
  skipped: number;
}

File: backend/app/models/feedback.py
from enum import StrEnum
from pydantic import BaseModel, Field


class EmailStatus(StrEnum):
    pending = "pending"
    sent = "sent"
    failed = "failed"


class FeedbackIn(BaseModel):
    meeting_id: str
    speaker_id: str
    comment: str = Field(max_length=5000)


class FeedbackUpdateIn(BaseModel):
    comment: str = Field(max_length=5000)


class FeedbackOut(BaseModel):
    id: str
    meeting_id: str
    speaker_id: str
    evaluator_id: str
    comment: str
    created_at: str
    updated_at: str


class EmailLogOut(BaseModel):
    id: str
    meeting_id: str
    speaker_id: str
    sent_at: str | None
    status: EmailStatus
    error_message: str | None


class EmailDispatchResult(BaseModel):
    sent: int
    failed: int
    skipped: int
